#Hypothesis Testing

1. State the Hypotheses Statement


We are testing whether the observed weekly operating costs are higher than the  model suggests. The hypotheses are:

.Null Hypothesis (H₀): The weekly operating cost follows the model.

𝐻
0
:
𝜇
=
1
,
000
+
5
⋅
𝑋

 Since x is given 600. We substitue the value of x.

  μ=1,000+5*600

  μ=4,000

  .Alternative Hypothesis (H₁): The weekly operating cost is higher than the model.

H
1
​
 :μ>4,000

This is a one-tailed test, where we are checking if the actual mean is greater than the mean given by the model.



2. Calculate the Test Statistic

We are given:

Sample mean weekly cost
xˉ=3,050


Theoretical mean weekly cost
μ 0=4,000


Standard deviation
σ=5⋅25=125

Sample size
n=25

In [ ]:
import math
from scipy.stats import norm
import numpy as np


In [ ]:
sample_mean = 3050  # Sample mean weekly cost (Rs.)
model_mean = 4000  # model mean weekly cost (Rs.)
std_dev = 5 * 25  # Standard deviation (Rs.)
sample_size = 25  # Sample size

Since, the population standard deviation is known to us. We will use the z-test formula.

z=(sample mean- model mean)/(population standard deviation/ (sample size)^2
z=(3050-4000)/(5*25/(25)^2)

In [ ]:
Z = (sample_mean - model_mean) / (std_dev / math.sqrt(sample_size))
print("Test Statistic (Z):", Z)

Test Statistic (Z): -38.0


3. Determine the Critical Value

For right tailed test with
α=0.05, we will look up the critical value from the Z-distribution table.

The critical value for
α=0.05 is:

In [ ]:
alpha = 0.05
Z_critical = norm.ppf(1 - alpha)
print("Critical Value (Z):", np.round(Z_critical, 2))

Critical Value (Z): 1.64


Z
critical=1.645

4. Make a Decision
We compare the calculated test statistic with the critical value:

If Z > Zcritical, we reject the null hypothesis.


If Z ≤ Zcritical, we fail to reject the null hypothesis.

In this case, the test statistic
Z=−38 is much less than the critical value
Zcritical=1.645. Therefore, we fail to reject the null hypothesis.

In [ ]:
if Z > Z_critical:
    decision = "Reject the null hypothesis"
else:
    decision = "Fail to reject the null hypothesis"

In [ ]:
print(f"Test Statistic (Z): {Z:.4f}")
print(f"Critical Value (Z): {Z_critical:.4f}")
print(f"Decision: {decision}")

Test Statistic (Z): -38.0000
Critical Value (Z): 1.6449
Decision: Fail to reject the null hypothesis


5. Conclusion

After running the hypothesis test, we found that the test statistic was **-38**, which is way smaller than the critical value of **1.645**. This essentially means that the sample's weekly operating costs are significantly lower than what the original model predicted (which was $4,000).

Because our test statistic is so far below the critical value, we **don’t have enough evidence** to support the restaurant owners' claim that their costs are higher than expected. In fact, it looks like the operating costs are **actually lower** than what the model suggested, based on this data.

So, the conclusion here is: the restaurant owners’ concern about higher costs doesn’t seem to be backed up by the numbers. The costs are likely in line with or even below what the model predicted.

# Chi Square

1. State the Hypotheses:

Null Hypothesis (H₀):
The two variables are independent , i.e there's no significant association between the type of device purchased and the customer's satisfaction level.

Alternative Hypothesis (H₁):
The two variables are dependent , i.e there's a significant association between the type of device purchased and the customer's satisfaction level

2. Compute the Chi-Square Statistic:
The Chi-Square test statistic is calculated using the formula:

    𝜒2=∑(𝑂−𝐸)2/E

.O is the observed frequency.


.E is the expected frequency, which is calculated by:

    E=(Total of observations (row) * Total of observations (columns)/Total number of observations

  The next step is to compute the expected frequencies for each cell in the contingency table and then calculate the Chi-Square statistic.

In [ ]:
import numpy as np
from scipy.stats import chi2_contingency


In [ ]:
observed = np.array([
    [50, 70],  # Very Satisfied
    [80, 100], # Satisfied
    [60, 90],  # Neutral
    [30, 50],  # Unsatisfied
    [20, 50]   # Very Unsatisfied
])

In [ ]:
chi2_stat, p_val, dof, expected = chi2_contingency(observed)

3. Determine the Critical Value:

.Significance Level (α): 0.05

.Degrees of Freedom (df): The degrees of freedom for a Chi-Square test of independence is given by:

    df=(row−1)×(column−1)
    
We will use the chi-square distribution to determine the critical value for df at a significance level of 0.05.




In [ ]:
alpha = 0.05

4. Make a Decision:
After computing the Chi-Square statistic, compare it to the critical value obtained from the Chi-Square distribution table. If the Chi-Square statistic is greater than the critical value, you reject the null hypothesis

In [ ]:
if p_val < alpha:
   decision="Reject the null hypothesis"
else :
  decision="Fail to reject the null hypothesis"

In [ ]:
print(f"Chi-Square Statistic: {chi2_stat:.4f}")
print(f"P-Value: {p_val:.4f}")
print(f"Degrees of Freedom: {dof}")
print(f"Decision: {decision}")
print(f"Expected Frequencies: \n{expected}")

Chi-Square Statistic: 5.6382
P-Value: 0.2278
Degrees of Freedom: 4
Decision: Fail to reject the null hypothesis
Expected Frequencies: 
[[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


Conclusion:
Since the p-value (0.2278) is greater than 0.05, we fail to reject the null hypothesis. This means that there is no significant association between the type of device purchased (Smart Thermostat vs. Smart Light) and customer satisfaction level. Therefore, customer satisfaction seems to be independent of the type of device purchased.